### ISSUE TITLE
Add machine readable --output options to `kubectl api-resources`

### ISSUE THREAD

**Issue by grant0417** [kubernetes/kubernetes#116230]
### What would you like to be added?

Currently the `kubectl api-resources` command `--output` flag only takes `wide` or `name`, a machine readable format such as `json` would be useful for scripting/utilizing `kubectl` 

### Why is this needed?

I am a dev at [fig](https://github.com/withfig) and help maintain the [autocomplete repo](https://github.com/withfig/autocomplete), we generate completions for the `kubectl` CLI based on `kubectl api-resources --output name` right now, but this lacks support for short names which are hard to parse from the human readable `kubectl api-resources` be parsed from a `kubectl api-resources --output json` command.

This might be a much less common use case for others but it still feels useful to have machine readable output for as many `--output` flags as possible.

**Comment by XDRAGON2002** [comment]
This sound like a decent issue to start with as a new contributor I feel? Could I take this up if it gets accepted by a sig?
